app/src/main/java/com/graeberj/taskman/core/data/preferences/DefaultPreferences.kt

```diff
@@ -0,0 +1,40 @@
+package com.graeberj.taskman.core.data.preferences
+
+import android.content.SharedPreferences
+import com.graeberj.taskman.auth.domain.models.LoggedInUser
+import com.graeberj.taskman.core.domain.preferences.Preferences
+
+class DefaultPreferences(
+    private val sharedPreferences: SharedPreferences
+) : Preferences {
+    override fun saveToken(token: String) {
+        sharedPreferences.edit().putString(Preferences.KEY_TOKEN, token).apply()
+    }
+
+    override fun saveFullName(name: String) {
+        sharedPreferences.edit().putString(Preferences.KEY_NAME, name).apply()
+    }
+
+    override fun saveUserId(userId: String) {
+        sharedPreferences.edit().putString(Preferences.KEY_USERID, userId).apply()
+    }
+
+    override fun loadUser(): LoggedInUser? {
+        val token = sharedPreferences.getString(Preferences.KEY_TOKEN, "") ?: ""
+        val userid = sharedPreferences.getString(Preferences.KEY_USERID, "") ?: ""
+        val fullName = sharedPreferences.getString(Preferences.KEY_NAME, "") ?: ""
+        if (token.isBlank() || userid.isBlank() || fullName.isBlank()) {
+            return null
+        }
+        return LoggedInUser(
+            token = token,
+            userId = userid,
+            fullName = fullName
+        )
+    }
```

Comment:
Now I see that you also modeled the behavior I commented before, just that you check for empty fields. Why not check for nulls instead and keep the default values of the shared pref fields at null? 
In the end, that's exactly why we have nulls - to indicate a value doesn't exist.